front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -1267,6 +1138,222 @@ export default function AssistantBuilder({
+      })(builderState.actionMode);
+  }
+
+  const body: t.TypeOf<typeof PostOrPatchAgentConfigurationRequestBodySchema> =
+    {
+      assistant: {
+        name: removeLeadingAt(builderState.handle),
+        pictureUrl: builderState.avatarUrl,
+        description: builderState.description.trim(),
+        status: isDraft ? "draft" : "active",
+        scope: builderState.scope,
+        action: actionParam,
+        generation: {
+          prompt: builderState.instructions.trim(),
+          model: builderState.generationSettings.modelSettings,
+          temperature: builderState.generationSettings.temperature,
+        },
+      },
+    };
+
+  const res = await fetch(
+    !agentConfigurationId
+      ? `/api/w/${owner.sId}/assistant/agent_configurations`
+      : `/api/w/${owner.sId}/assistant/agent_configurations/${agentConfigurationId}`,
+    {
+      method: !agentConfigurationId ? "POST" : "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    }
+  );
+
+  if (!res.ok) {
+    throw new Error("An error occurred while saving the configuration.");
+  }
+
+  const newAgentConfiguration: {
+    agentConfiguration: LightAgentConfigurationType | AgentConfigurationType;
+  } = await res.json();
+  const agentConfigurationSid = newAgentConfiguration.agentConfiguration.sId;
+
+  // PATCH the linked slack channels if either:
+  // - there were already linked channels
+  // - there are newly selected channels
+  // If the user selected channels that were already routed to a different assistant, the current behavior is to
+  // unlink them from the previous assistant and link them to the this one.
+  if (
+    selectedSlackChannels.length ||
+    slackChannelsLinkedWithAgent.filter(
+      (channel) => channel.agentConfigurationId === agentConfigurationId
+    ).length
+  ) {
+    const slackLinkRes = await fetch(
+      `/api/w/${owner.sId}/assistant/agent_configurations/${agentConfigurationSid}/linked_slack_channels`,
+      {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          slack_channel_ids: selectedSlackChannels.map(
+            ({ slackChannelId }) => slackChannelId
+          ),
+        }),
+      }
+    );
+
+    if (!slackLinkRes.ok) {
+      throw new Error("An error occurred while linking Slack channels.");
+    }
+  }
+
+  return newAgentConfiguration.agentConfiguration;
+}
+
+function TryModalInBuilder({
+  owner,
+  builderState,
+  disabled,
+}: {
+  owner: WorkspaceType;
+  builderState: AssistantBuilderState;
+  disabled: boolean;
+}) {
+  const { user } = useUser();
+  const [assistant, setAssistant] = useState<
+    LightAgentConfigurationType | AgentConfigurationType | null
+  >(null);
+
+  async function onTryClick() {
+    setAssistant(
+      await submitForm({
+        owner,
+        builderState: { ...builderState },
+        agentConfigurationId: null,
```

Comment:
Can we add a comment on this line to explain that this is the "hack" to create a new assistant everytime a user clicks on the try button?